File: branches/Medusa/Source/Fantasy.Jobs/TaskItemAggregateReader.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;


namespace Fantasy.Jobs
{
    public class TaskItemAggregateReader : ObjectWithSite, ITagValueProvider
    {
        #region ITagValueProvider Members

        public char Prefix
        {
            get { return '@'; }
        }

        public string GetTagValue(string tag, IDictionary<string, object> context)
        {
            string[] expr = tag.Split(new char[] {'.'}, 2, StringSplitOptions.RemoveEmptyEntries);

            IJob job = (IJob)this.Site.GetService(typeof(IJob));
            
            TaskItem[] items = job.GetEvaluatedItemsByCatetory(expr[0]);

            IEnumerable<string> values;
            if(expr.Length > 1)
            {
                values = items.Select(x => x[expr[1]]);
            }
            else
            {
                values = items.Select(x=>x.Name);
            }

            return String.Join(";", values.ToArray());
        }

        public bool HasTag(string tag, IDictionary<string, object> context)
        {
            return true;
        }

        public bool IsEnabled(IDictionary<string, object> context)
        {
            return (bool)context.GetValueOrDefault("EnableTaskItemReader", true) &&
                this.Site != null && this.Site.GetService(typeof(IJob)) != null;
        }

        #endregion
    }
 
 
}

File: AutoConverted/Fantasy.Jobs/TaskItemAggregateReader.java
package Fantasy.Jobs;

public class TaskItemAggregateReader extends ObjectWithSite implements ITagValueProvider
{
//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region ITagValueProvider Members

	public final char getPrefix()
	{
		return '@';
	}

	public final String GetTagValue(String tag, java.util.Map<String, Object> context)
	{
		String[] expr = tag.split(new char[] {'.'}, 2, StringSplitOptions.RemoveEmptyEntries);

		IJob job = (IJob)this.Site.GetService(IJob.class);

		TaskItem[] items = job.GetEvaluatedItemsByCatetory(expr[0]);

		Iterable<String> values;
		if(expr.length > 1)
		{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
			values = items.Select(x => x[expr[1]]);
		}
		else
		{
//C# TO JAVA CONVERTER TODO TASK: Lambda expressions and anonymous methods are not converted by C# to Java Converter:
			values = items.Select(x=>x.getName());
		}

		return DotNetToJavaStringHelper.join(";", values.toArray());
	}

	public final boolean HasTag(String tag, java.util.Map<String, Object> context)
	{
		return true;
	}

	public final boolean IsEnabled(java.util.Map<String, Object> context)
	{
		return (boolean)context.GetValueOrDefault("EnableTaskItemReader", true) && this.Site != null && this.Site.GetService(IJob.class) != null;
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}